### PR TITLE
feat: mobile filters menu

### PR DESCRIPTION
**Pull request by cbachmeier**
- Adds mobile filter menu
- Make the Marketplace dropdown styles more re-usable
- Add Sort dropdown to mobile filter menu
- Update display text for sort drop down and remove icons

Misc:
- Allow sweep with buyNow toggled off (prod request)

### CODE REVIEW

File: src/nft/components/common/SortDropdown/FilterSortDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      <Box
+        as="img"
+        alt={dropDownOption.displayText}
+        width="20"
+        height="20"
+        objectFit="cover"
+        src="/nft/svgs/checkmark.svg"
+      />
+    ) : (
+      <></>
```
Comment (aballerr) [inline src/nft/components/common/SortDropdown/FilterSortDropdown.tsx:39]: probably can just pull null here

File: src/nft/components/common/SortDropdown/FilterSortDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  console.log(sortBy, dropDownOption.sortBy)
+  const checkMark =
+    dropDownOption.sortBy !== undefined && sortBy === dropDownOption.sortBy ? (
+      <Box
+        as="img"
+        alt={dropDownOption.displayText}
+        width="20"
+        height="20"
+        objectFit="cover"
+        src="/nft/svgs/checkmark.svg"
```
Comment (aballerr) [inline src/nft/components/common/SortDropdown/FilterSortDropdown.tsx:36]: is it possible to import this source?

File: src/nft/components/collection/MarketplaceSelect.tsx [hunk 3]
```diff
@@ -22,6 +22,39 @@ export const MARKETPLACE_ITEMS = {
...
+      overflowX={'hidden'}
+      overflowY={'hidden'}
+      fontWeight="normal"
+      className={`${subheadSmall} ${styles.subRowHover}`}
+      paddingLeft="12"
+      paddingRight="16"
+      borderRadius="12"
+      cursor="pointer"
+      maxHeight="44"
+      style={{ paddingBottom: '22px', paddingTop: '22px' }}
```
Comment (aballerr) [inline src/nft/components/collection/MarketplaceSelect.tsx:47]: we should probably make this a styled-component

File: src/nft/components/collection/CollectionNfts.tsx [hunk 4]
```diff
@@ -265,51 +296,14 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
     setIsCollectionNftsLoading(isLoadingNext)
   }, [isLoadingNext, setIsCollectionNftsLoading])
 
-  const hasRarity = getRarityStatus(rarityStatusCache, collectionStats?.address, collectionNfts)
+  const hasRarity = useMemo(() => {
+    const hasRarity = getRarityStatus(rarityStatusCache, collectionStats?.address, collectionNfts) ?? false
+    setHasRarity(hasRarity)
+    return hasRarity
```
Comment (aballerr) [inline src/nft/components/collection/CollectionNfts.tsx:302]: what's the point of returning rarity here and then also setting rarity?  Wouldn't it make sense to just get the rarity from filters collection only or not set it at all

File: src/nft/pages/collection/index.tsx [hunk 5]
```diff
@@ -110,8 +131,29 @@ const Collection = () => {
                 />
               </CollectionDescriptionSection>
               <CollectionDisplaySection>
-                <Box position="sticky" top="72" width="0">
-                  {isFiltersExpanded && <Filters traitsByGroup={collectionStats?.traits ?? {}} />}
+                <Box
```
Comment (aballerr) [inline src/nft/pages/collection/index.tsx:134]: We should use a styled-component here

File: src/nft/utils/scrollToTop.ts [hunk 6]
```diff
@@ -1,5 +1,10 @@
+import { getIsMobile } from 'nft/hooks'
+
 export const scrollToTop = () => {
-  window.document.getElementById('nft-anchor')?.scrollIntoView({
+  const isMobile = getIsMobile()
```
Comment (aballerr) [inline src/nft/utils/scrollToTop.ts:4]: This is on me, mobile wasn't scrolling to a good spot, so I fixed it

**Review by aballerr — COMMENTED** [review]
I'm not sure why, but it's possible to scroll the content behind the filters while they are open.  I was able to do this when scrolling down on web and also on my phone.  It's especially hard when trying to scroll at the bottom of filters

File: src/nft/pages/collection/index.tsx [hunk 7]
```diff
@@ -30,6 +32,24 @@ const CollectionDisplaySection = styled(Row)`
...
+  opacity: 1;
+  transition: 125ms ease opacity;
+  :hover {
+    opacity: 0.6;
+  }
+  :active {
+    opacity: 0.4;
+  }
+`
+
```
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/nft/pages/collection/index.tsx:52]: Updated here and where I got the codeblock in BagHeader

File: src/nft/components/common/SortDropdown/FilterSortDropdown.tsx (hunk 2 above)
Comment (cbachmeier) [reply src/nft/components/common/SortDropdown/FilterSortDropdown.tsx:36]: We haven't previously imported svgs from the interface public repo. Previously examples in the ListPage, CollectionStats, and ProfileAssetRows

File: src/nft/components/collection/CollectionNfts.tsx (hunk 4 above)
Comment (cbachmeier) [reply src/nft/components/collection/CollectionNfts.tsx:302]: this is where the value is retrieved from the queried collection and set in the hook

File: src/nft/components/common/SortDropdown/FilterSortDropdown.tsx (hunk 1 above)
Comment (cbachmeier) [reply src/nft/components/common/SortDropdown/FilterSortDropdown.tsx:39]: I can't pass null as a React element

**Review by aballerr — COMMENTED** [review]
I still can scroll the background on my phone

**Review by aballerr — APPROVED** [review]

**Pull request merged**